### ISSUE TITLE
Implement di-order and di-unused for referenced functions

### ISSUE THREAD

**Issue by remcohaszing** [Gillespie59/eslint-plugin-angular#216]
`di-order` and `di-unused` check for the following syntax.
```js
factory('myService', function($http, $q) {});
```

These rules should also check for functions passed by reference.
```js
factory('myService', myService);

function myService($http, $q) {}
```

**Comment by maksimr** [comment]
+1

Also di-unused does not work with array notation `['$log', function($log){...}]`

**Comment by remcohaszing** [comment]
#283 doesn't fix this issue, but it does issue no longer exists in the development branch.
